src/components/navigation-header/components/product-page-content/index.tsx

```diff
@@ -40,22 +48,56 @@ const ProductPageHeaderContent = () => {
   const currentProduct = useCurrentProduct()
   const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
   const isBetaProduct = useIsBetaProduct(currentProduct.slug)
-  const betaProductSlugs = __config.dev_dot.beta_product_slugs
+  const productLogo = PRODUCT_SLUGS_TO_LOGOS[currentProduct.slug]
+  const isProductHomePage = currentPath === `/${currentProduct.slug}`
   const companyLogo = (
     <InlineSvg className={s.companyLogo} src={HashiCorpLogo} />
   )
-  const homeMenuItem = { icon: 'home', label: 'Developer Home', path: '/' }
-  const betaProductMenuItems = betaProductSlugs.map((slug: ProductSlug) => ({
-    icon: slug,
-    label: productSlugsToNames[slug],
-    path: `/${slug}`,
-  }))
+
+  // Build menu items
+  const betaProductItems = []
+  const comingSoonProductItems = []
+  Object.keys(productSlugsToNames).forEach((productSlug: ProductSlug) => {
+    // Exclude Sentinel for now
+    if (productSlug === 'sentinel') {
+      return
+    }
+
+    // Generate properties of each menu item
+    const icon = productSlug
+    const label = productSlugsToNames[productSlug]
+    const path = `/${productSlug}`
+
+    // Push the menu item to the correct array
+    if (getIsBetaProduct(productSlug)) {
+      betaProductItems.push({
+        badge: {
+          text: 'Beta',
+          color: 'highlight',
+        },
+        icon,
+        label,
+        path,
+      })
+    } else {
+      comingSoonProductItems.push({
+        ariaLabel: `Coming soon: ${label}`,
+        icon,
+        label,
+      })
+    }
+  })
+  const homeMenuItem = {
+    // TODO as string was not accepted
+    icon: 'home' as $TSFixMe,
```

Comment:
`as const` might work if its a union of specific strings

Reply:
Might be able to resolve in #588 where I'm working on an abstraction, but if that doesn't work I'll try this! Going to link to this thread in that PR so I don't forget. 📌